### PR TITLE
feat: Add keyMatch4

### PR DESCRIPTION
**Pull request by Edmond-J-A**
Signed-off-by: Edmond-J-A <edomondja@gmail.com>

### CODE REVIEW

File: src/util/BuiltInFunctions.lua [hunk 1]
```diff
@@ -101,6 +101,39 @@ function BuiltInFunctions.keyMatch3(key1, key2)
 	return BuiltInFunctions.regexMatch(key1, "^"..key.."$")
 end
 
+function BuiltInFunctions.keyMatch4Func(args)
```
Comment (rushitote) [inline src/util/BuiltInFunctions.lua:104]: Add this comment for the function:
```lua
-- Wrapper for keyMatch4
```

File: src/util/BuiltInFunctions.lua [hunk 2]
```diff
@@ -101,6 +101,39 @@ function BuiltInFunctions.keyMatch3(key1, key2)
...
+    BuiltInFunctions.validateVariadicArgs(2, args)
+    return BuiltInFunctions.keyMatch4(args[1], args[2])
+end
+
+function BuiltInFunctions.keyMatch4(key1, key2)
```
Comment (rushitote) [inline src/util/BuiltInFunctions.lua:109]: Add this comment too:
```lua
-- KeyMatch4 determines whether key1 matches the pattern of key2 (similar to RESTful path), key2 can contain a *.
-- Besides what KeyMatch3 does, KeyMatch4 can also match repeated patterns:
-- "/parent/123/child/123" matches "/parent/{id}/child/{id}"
-- "/parent/123/child/456" does not match "/parent/{id}/child/{id}"
-- But KeyMatch3 will match both.
```

**Review by rushitote — CHANGES_REQUESTED** [review]
Great work!

File: src/util/BuiltInFunctions.lua (hunk 1 above)
Comment (Edmond-J-A) [reply src/util/BuiltInFunctions.lua:105]: Got it ,fixed

**Review by rushitote — APPROVED** [review]

**Pull request merged**